### PR TITLE
Add docutils definitions and (incomplete) modules

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
python/mypy#10242, cc @rchen152: Could you have a look at the pytype errors? I can't make sense of them. I suspect this is also related to using `tuple[X, ...]`, but I'm not sure.

**Comment by rchen152** [comment]
The pytype test failures are caused by pytype assuming that some imports are modules when they're actually classes. I filed https://github.com/google/pytype/issues/882. You should be able to work around this by modifying the imports that pytype is unhappy with like this:

```
stubs/docutils/docutils/frontend.pyi
------------------------------------
-from . import SettingsSpec
+from docutils import SettingsSpec
 
stubs/docutils/docutils/io.pyi
------------------------------
-from . import TransformSpec
+from docutils import TransformSpec
 
 stubs/docutils/docutils/parsers/__init__.pyi
---------------------------------------------
-from .. import Component
+from docutils import Component

stubs/docutils/docutils/parsers/rst/__init__.pyi
--------------------------------------------------
-from ... import parsers
+from docutils import parsers
```

File: stubs/docutils/docutils/io.pyi [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+class FileInput(Input):
+    def __init__(
+        self,
+        source: Optional[Any] = ...,
+        source_path: Optional[Any] = ...,
+        encoding: Optional[str] = ...,
+        error_handler: str = ...,
+        autoclose: bool = ...,
+        mode: Literal["r", "rb"] = ...,
```
Comment (JelleZijlstra) [inline stubs/docutils/docutils/io.pyi:40]: This is the default but not the only possibility; it also allows `rU` in Python 2 at least.

File: stubs/docutils/docutils/io.pyi [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    ) -> None: ...
+    def readlines(self) -> list[str]: ...
+    def close(self) -> None: ...
+
+class FileOutput(Output):
+    mode: ClassVar[Literal["w", "wb"]]
```
Comment (JelleZijlstra) [inline stubs/docutils/docutils/io.pyi:46]: This can also be wider; I guess we can use the aliases from `_typeshed`

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Thanks! Just two comments

**Pull request merged**